libs/dal/src/repositories/subscriber/subscriber.repository.ts

```diff
@@ -10,6 +10,8 @@ import type { EnforceEnvOrOrgIds } from '../../types';
 import { EnvironmentId, ISubscribersDefine, OrganizationId } from '@novu/shared';
 
 type SubscriberQuery = FilterQuery<SubscriberDBModel> & EnforceEnvOrOrgIds;
+type SubscriberDeleteQuery = Pick<SubscriberQuery, 'subscriberId' | '_environmentId'> & EnforceEnvOrOrgIds;
+type SubscriberDeleteManyQuery = Pick<SubscriberQuery, 'subscriberId' | '_id' | '_environmentId'> & EnforceEnvOrOrgIds;
```

Comment:
i decided to create those types because, without those types, the client of the methods could think that other query params are in use in the query while they are not at the moment.